### PR TITLE
aks-preview: Add podMaxPids argument for node config

### PR DESCRIPTION
**Pull request by justindavies**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

### CODE REVIEW

**Comment by yonzhan** [comment]
aks-preview

File: src/aks-preview/azext_aks_preview/custom.py
Comment (Juliehzl) [inline src/aks-preview/azext_aks_preview/custom.py:3881]: how do customer use the feature in cli?

Comment (justindavies) [reply src/aks-preview/azext_aks_preview/custom.py:3881]: They would create a node config or kubelet config json file and apply it to the cluster as documented here: https://docs.microsoft.com/en-us/azure/aks/custom-node-configuration

**Review by Juliehzl — APPROVED** [review]
LGTM, but please also get approval from your team member.

**Comment by Juliehzl** [comment]
Hi @justindavies , please resolve conflicts and get approval from code owner. Then I could merge the PR.

**Comment by FumingZhang** [comment]
Refer to the new PR [#3588](https://github.com/Azure/azure-cli-extensions/pull/3588).

**Pull request closed without merging**
